examples/stable-diffusion/text_to_image_generation.py

```diff
@@ -295,32 +275,82 @@ def main():
 
     # Generate images
     if args.control_image is not None:
+        from diffusers import ControlNetModel
+
         model_dtype = torch.bfloat16 if args.bf16 else None
         controlnet = ControlNetModel.from_pretrained(args.controlnet_model_name_or_path, torch_dtype=model_dtype)
-        pipeline = GaudiStableDiffusionControlNetPipeline.from_pretrained(
-            args.model_name_or_path,
-            controlnet=controlnet,
-            **kwargs,
-        )
-
-        # Set seed before running the model
-        set_seed(args.seed)
 
-        outputs = pipeline(
-            prompt=args.prompts,
-            image=control_image,
-            num_images_per_prompt=args.num_images_per_prompt,
-            batch_size=args.batch_size,
-            num_inference_steps=args.num_inference_steps,
-            guidance_scale=args.guidance_scale,
-            negative_prompt=args.negative_prompts,
-            eta=args.eta,
-            output_type=args.output_type,
-            profiling_warmup_steps=args.profiling_warmup_steps,
-            profiling_steps=args.profiling_steps,
-            **res,
-        )
+        if sdxl:
+            from optimum.habana.diffusers import GaudiStableDiffusionXLInstantIDPipeline
```

Comment:
Hmm we should not hijack this case strictly for InstantID sample.   The standard `GaudiStableDiffusionXLControlNetPipeline` should be default here, so please adjust logic.   See example here:
https://github.com/dsocek/optimum-habana/commit/64e56de700202ffb254c0d033fe7a7637b9a8211

Basically need something like:
```
if sdxl and controlnet:
  if instantid:
    # InstantID pipe 
 else:
    # default SDXL ControlNet pipe
```

Reply:
I've added `--ip_adapter_path` argument that allows to delineate the logic. If this argument is specified, `GaudiStableDiffusionXLInstantIDPipeline` is used

Reply:
done